src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -75,16 +75,24 @@ public User createUser(User newUser) {
         return newUser;
     }
 
-    public User updateUser(Long id, UserUpdateDTO userUpdateDTO, String updatingUsername) {
+    public User updateUser(Long id, User userInput, String updatingUsername) {
```

Comment:
We shoud be able to remove all these useless arguments. 
```suggestion
    public User updateUser(UserUpdateDTO userUpdateDTO, Principal principal) {
```
and just find the user by calling `this.userRepository.findByUsername(principal.getName()).orElse(null)`